app/download/views.py

```diff
@@ -0,0 +1,23 @@
+from flask import Blueprint, jsonify, make_response, request, send_file
+
+from app import document_store
+
+from app.utils.store import DocumentStoreError
+
+download_blueprint = Blueprint('download', __name__, url_prefix='')
+
+
+@download_blueprint.route('/services/<uuid:service_id>/documents/<uuid:document_id>', methods=['GET'])
+def download_document(service_id, document_id):
+    if 'key' not in request.args:
+        return jsonify(error='Missing decryption key'), 400
+
+    try:
+        document = document_store.get(service_id, document_id, request.args.get('key'))
+    except DocumentStoreError as e:
+        return jsonify(error=str(e)), 400
```

Comment:
We have those in the request URL. This will also probably change once we add a user-facing HTML page, so I reckon we'll revisit this once we know what information that page needs.